python/tk_maya/menu_generation.py

```diff
@@ -294,6 +294,20 @@ def add_command_to_menu(self, menu):
             
         pm.menuItem(**params)
         
+    def __exception_trap_callback_wrapper(self, callback):
```

Comment:
This looks like a pattern that should be made available in the base engine class. As a decorator perhaps?